pkg/azuredx/tokenprovider/token_provider_test.go

```diff
@@ -0,0 +1,105 @@
+package tokenprovider
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/stretchr/testify/require"
+)
+
+const TestClientId = "3c882424-c381-45c0-a123-0a62ebc989dc"
+const TestTenantId = "b931e0dd-f42c-444a-83a9-63de6f2d771a"
+const TestClientSecret = "3093e953-91e1-48f4-9ad0-e8f750c38f93"
+const Scope = "https://kusto.kusto.windows.net/.default"
+const ExpectedToken string = "4cb83b87-0ffb-4abd-82f6-48a8c08afc53"
+
+type tokenCacheFake struct{}
+
+func (c *tokenCacheFake) GetAccessToken(_ context.Context, credential TokenCredential, scopes []string) (string, error) {
+	err := credential.Init()
+	if err != nil {
+		return "", err
+	}
+	return ExpectedToken, nil
+}
+
+func (c *tokenCacheFake) Purge() {
+	panic("not implemented")
+}
+
+func TestTokenProvider(t *testing.T) {
+	ctx := context.Background()
+
+	newClientSecretCredentialOriginal := newClientSecretCredential
```

Comment:
Having a package variable for the test is a bit risky I would move it to an interface and either pass it as an argument in the `Init` method or store it in the `clientSecretCredential` but up to you if you don't think it's worth it.

Reply:
yeah that's true. hmm at the moment, the token cache is generic and I'd like to avoid adding a dependency to azidentity there. think I'll need to keep it as is for now and revisit this later when there's more time. thanks for good feedback!